src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -142,6 +142,33 @@ private void LogSuccessfulTelemetryToClient(AuthenticationResult authenticationR
                 telemetryEventDetails.SetProperty(TelemetryConstants.PopToken, authenticationResult.TokenType.Equals(Constants.PoPTokenType));
                 telemetryEventDetails.SetProperty(TelemetryConstants.RemainingLifetime, (authenticationResult.ExpiresOn - DateTime.Now).TotalMilliseconds);
                 telemetryEventDetails.SetProperty(TelemetryConstants.ActivityId, authenticationResult.CorrelationId);
+                telemetryEventDetails.SetProperty(TelemetryConstants.RefreshOn, 
+                    authenticationResult.AuthenticationResultMetadata.RefreshOn.HasValue ?
+                    DateTimeHelpers.DateTimeToUnixTimestampMilliseconds(authenticationResult.AuthenticationResultMetadata.RefreshOn.Value)
+                    : 0);
+
+                LogCacheDetailsToTelemetryClient(cacheDetails, telemetryEventDetails);
+            }
+        }
+
+        private void LogCacheDetailsToTelemetryClient(Dictionary<string, object> cacheDetails, MsalTelemetryEventDetails telemetryEventDetails)
+        {
+            foreach (var cacheDetail in cacheDetails)
```

Comment:
perf: enumerating in dictionary is O(n) operation. Better to use typical dictionary operations which are O(1)

See `TryGetValue` in particular.